Program.cs

```diff
@@ -1,10 +1,69 @@
-﻿namespace WeatherMonitoringSystem
+﻿using WeatherMonitoringSystem.model;
+using WeatherMonitoringSystem.Observer;
+using WeatherMonitoringSystem.Strategy;
+using WeatherMonitoringSystem.utils;
+
+namespace WeatherMonitoringSystem
 {
     internal class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            Console.WriteLine("Weather Monitoring System");
+
+            try
+            {
+                var bots = FileConfig();
+               
+                WeatherSubject subject = new WeatherSubject();
+
+                AddBotObserversTOSubject(subject, bots);
+
+                WeatherData weather = UserInput();
+
+                subject.SetWeatherValue(weather);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
         }
+
+        public static IEnumerable<Bot> FileConfig()
+        {
+            ConfigurationsFileReader configFileReader = new ConfigurationsFileReader();
+
+            return configFileReader.Read(FilePath.filePath);
+        }
+        public static void AddBotObserversTOSubject(IWeatherSubject subject, IEnumerable<Bot> bots)
+        {
+            RainBot rainBot = (RainBot)bots.FirstOrDefault(bot => bot.GetType().Name is "RainBot");
```

Comment:
```suggestion
            RainBot rainBot = (RainBot)bots.FirstOrDefault(bot => bot.GetType() == typeof(RainBot));
```

That way when you change the RainBot class name you don't have to look for string usages and change them, the IDE would be able to dot the refactoring in all usages.